schema/sourcecode.md

```diff
@@ -1,87 +1,131 @@
 
-# Software Source Code
+# Software Source Code Metadata
 
-This document outlines the *required* and *recommended* properties for
-describing source code.
+To classify a record as a computer programming source code, `"@type: "SoftwareSourceCode"` should be used in the json schema. This will classify the record such as compile ready solutions, code snippet samples, scripts, etc. as a specific Schema.Org type called `SoftwareSourceCode` for which the metadata should be described using the [core metadata](https://github.com/I-GUIDE/data-catalog/blob/main/schema/core.md), as well as the software-source-code-specific properties for the [Schema:SoftwareSourceCode](https://schema.org/SoftwareSourceCode) class. The following table outlines the required and optional properties selected from Schema.Org vocabulary to design the I-GUIDE software source code metadata schema. These properties are encoded as `1` or `1+` for **required** and `0,1` or `0+` for **optional** in the Cardinality column of the table below.
 
-While all Schema.org properties are valid, below is a suggested subset to
-represent a single source code object using the 
-*[SoftwareSourceCode](https://schema.org/SoftwareSourceCode)* class.
+|Property|Class|Expected Type|Cardinality|Description|
+|---|---|---|---|---|
+| [programmingLanguage](https://schema.org/programmingLanguage) | SoftwareSourceCode | ComputerLanguage \| Text | 1+ | The computer programming language |
+| [runtimePlatform](https://schema.org/runtimePlatform) | SoftwareSourceCode | Text | 1+ | Runtime platform or script interpreter dependencies |
+| [targetProduct](https://schema.org/targetProduct) | SoftwareSourceCode | [SoftwareApplication](https://github.com/I-GUIDE/data-catalog/blob/main/schema/softwareapp.md) | 1+ | Target Operating System or Product to which the code applies |
+| [codeRepository](https://schema.org/codeRepository) | SoftwareSourceCode | URL \| Text | 0,1 | Link to the repository where the un-compiled, human readable code and related code is located |
+| [codeSampleType](https://schema.org/codeSampleType) | Text | 0,1 | What type of code sample: full (compile-ready) solution, code snippet (incomplete), inline code, scripts, template |
+| [text](https://schema.org/text) | CreativeWork | Text | 0,1 | The textual content of the source code. |
```

Comment:
I'm not sure that we want to embed source code within the schema. I wonder if this could have security implications.